cukinia

```diff
@@ -621,6 +622,37 @@ _cukinia_mount()
 	return 0
 }
 
+# _cukinia_i2c: Check if i2c device exists
+# Optional: Check of a device address
+# arg1: the bus number
+# arg2: (optional) the device address
+_cukinia_i2c()
+{
+	local bus_number="$1"
+	local device_address="$2"
+	local ret
+
+	_cukinia_prepare "Checking if i2c-$bus_number${device_address:+-$device_address} exists"
+
+	if [ ! -d /sys/bus/i2c/devices/i2c-"$bus_number" ]; then
+		return 1
+	fi
+
+	if [ -z "$device_address" ]; then
+		return 0
+	fi
+
+	if [ ! -x "$(which i2cdetect)" ]; then
+		_cukinia_prepare "cukinia_i2c: i2cdetect is required"
+		return 1
+	fi
+
+	device_address=${device_address#0x}
+	i2cdetect -y "$bus_number" 0x"$device_address" 0x"$device_address" | cut -d ':' -f2 | grep "$device_address"
```

Comment:
I found the following issue, let's look at this example:
```
$ i2cdetect -y 7
     0  1  2  3  4  5  6  7  8  9  a  b  c  d  e  f
00:                         08 -- -- -- -- -- -- -- 
10: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
20: -- -- -- -- -- -- -- -- -- -- -- -- UU -- -- -- 
30: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
40: -- -- -- -- 44 -- -- -- -- -- -- -- -- -- -- -- 
50: UU -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
60: -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- 
70: -- -- -- -- -- -- -- --   
```

In this case device 2c shows up as `UU` because requested by a driver and wouldn't be matched by the grep above:

```
$ ls -l /sys/bus/i2c/devices/
lrwxrwxrwx. 1 root root 0 Oct  1 09:21 7-002c -> ../../../devices/...../i2c-7/7-002c
```